crates/rust-client/src/sync/mod.rs

```diff
@@ -162,10 +163,23 @@ impl<R: FeltRng> Client<R> {
         self.store.get_sync_height().await.map_err(|err| err.into())
     }
 
-    /// Syncs the client's state with the current state of the Miden network.
-    /// Before doing so, it ensures the genesis block exists in the local store.
+    /// Syncs the client's state with the current state of the Miden network. Returns the block
+    /// number the client has been synced to.
     ///
-    /// Returns the block number the client has been synced to.
+    /// The sync process is done in multiple steps:
+    /// 1. A request is sent to the node to get the state updates. This request includes tracked
+    ///    account IDs and the tags of notes that might have changed or that might be of interest to
+    ///    the client.
+    /// 2. A response is received with the current state of the network. The response includes
+    ///    information about new/committed/consumed notes, updated accounts, and committed
+    ///    transactions.
+    /// 3. Notes are updated with their new states.
+    /// 4. New notes are checked, only relevant notes are stored.
```

Comment:
```suggestion
    /// 4. New notes are checked, and only relevant are stored.
```
Also, what about mentioning the criteria used to evaluate which notes are kept?

Reply:
Agree that it would be nice to touch on the criteria used for keeping/updating entities. Also agree on the suggestion, but to nitpick a bit more I'd write it like this
```suggestion
    /// 4. New notes are checked, and only relevant ones are stored.
```